test/common/upstream/logical_dns_cluster_test.cc

```diff
@@ -389,8 +389,9 @@ TEST_F(LogicalDnsClusterTest, BadConfig) {
               port_value: 8000
   )EOF";
 
-  EXPECT_THROW_WITH_MESSAGE(setupFromV2Yaml(custom_resolver_yaml), EnvoyException,
-                            "LOGICAL_DNS clusters must NOT have a custom resolver name set");
+  EXPECT_THROW_WITH_MESSAGE(
```

Comment:
We need extra tests for dns_impl_test.cc as well as logical/strict DNS cluster tests that exercise the new code.